panic in binary.Unmarshal

### What did you do?

Wrote a simple fuzz target using https://blog.golang.org/fuzz-beta. Immediately found first crasher.

```
binary.Unmarshal([]byte("\x00\xff\x80"))
```

https://play.golang.org/p/41B5GBalYFb

### What did you expect to see?

An error.

### What did you see instead?

```
panic: runtime error: slice bounds out of range [:-1]
```

### Does this issue reproduce with the latest release?

Yes, I can reproduce the issue after upgrading to the latest commit.

```
go get github.com/Rhymen/go-whatsapp/binary@v0.1.2-0.20210714125832-447ed9cc6179
```

--- a/binary/decoder.go
+++ b/binary/decoder.go
@@ -109,7 +109,7 @@ func (r *binaryDecoder) readPacked8(tag int) (string, error) {
 		ret += lower + upper
 	}
 
-	if startByte>>7 != 0 {
+	if startByte>>7 != 0 && len(ret) != 0 {
 		ret = ret[:len(ret)-1]
 	}
 	return ret, nil

--- a/binary/node.go
+++ b/binary/node.go
@@ -1,10 +1,12 @@
 package binary
 
 import (
-	"fmt"
+	"errors"
 	pb "github.com/Rhymen/go-whatsapp/binary/proto"
 	"github.com/golang/protobuf/proto"
 )
+
+var ErrInvalidNode = errors.New("invalid Node")
 
 type Node struct {
 	Description string
@@ -42,7 +44,7 @@ func marshalMessageArray(messages []interface{}) ([]Node, error) {
 		} else {
 			ret[i], ok = m.(Node)
 			if !ok {
-				return nil, fmt.Errorf("invalid Node")
+				return nil, ErrInvalidNode
 			}
 		}
 	}

--- a/binary/node_test.go
+++ b/binary/node_test.go
@@ -1,6 +1,7 @@
 package binary
 
 import (
+	"errors"
 	"fmt"
 	"github.com/Rhymen/go-whatsapp/binary/proto"
 	"reflect"
@@ -70,4 +71,20 @@ func TestMarshal(t *testing.T) {
 		t.Errorf("content changed")
 		t.Fail()
 	}
+}
+
+func TestUnmarshal_InvalidNode(t *testing.T) {
+	invalidMsg := []byte("\x00\xff\x80")
+
+	gotNode, err := Unmarshal(invalidMsg)
+	if gotNode != nil {
+		t.Errorf("node is expected to be nil")
+		t.Fail()
+	}
+
+	if !errors.Is(err, ErrInvalidNode) {
+		t.Errorf("err is expected to be %s but got %s", ErrInvalidNode, err)
+		t.Fail()
+	}
+
 }

--- a/binary/node_test.go
+++ b/binary/node_test.go
@@ -86,5 +86,4 @@ func TestUnmarshal_InvalidNode(t *testing.T) {
 		t.Errorf("err is expected to be %s but got %s", ErrInvalidNode, err)
 		t.Fail()
 	}
-
 }

